packages/react-native-audio-api/common/cpp/audioapi/utils/RingBiDirectionalBuffer.hpp

```diff
@@ -0,0 +1,198 @@
+  bool popFront() noexcept(std::is_nothrow_destructible<T>::value) {
+    if (isEmpty()) [[ unlikely ]] {
+      return false;
+    }
+    buffer_[headIndex_].~T();
+    headIndex_ = nextIndex(headIndex_);
+    return true;
+  }
+
+  /// @brief Pop a value from the back of the buffer.
+  /// @param out The value popped from the buffer.
+  /// @return True if the value was popped successfully, false if the buffer is empty.
+  bool popBack(T& out) noexcept(std::is_nothrow_move_constructible<T>::value && std::is_nothrow_destructible<T>::value) {
+    if (isEmpty()) [[ unlikely ]] {
+      return false;
+    }
+    tailIndex_ = prevIndex(tailIndex_);
+    out = std::move(buffer_[tailIndex_]);
+    buffer_[tailIndex_].~T();
+    return true;
+  }
+
+  /// @brief Pop a value from the back of the buffer.
+  /// @return True if the value was popped successfully, false if the buffer is empty.
+  bool popBack() noexcept(std::is_nothrow_destructible<T>::value) {
+    if (isEmpty()) [[ unlikely ]] {
+      return false;
+    }
+    tailIndex_ = prevIndex(tailIndex_);
+    buffer_[tailIndex_].~T();
+    return true;
+  }
+
+  /// @brief Peek at the front of the buffer.
+  /// @return A const reference to the front element of the buffer.
+  const inline T& peekFront() const noexcept {
+    return buffer_[headIndex_];
+  }
+
+  /// @brief Peek at the back of the buffer.
+  /// @return A const reference to the back element of the buffer.
+  const inline T& peekBack() const noexcept {
+    return buffer_[prevIndex(tailIndex_)];
+  }
+
+  /// @brief Peek at the front of the buffer.
+  /// @return A mutable reference to the front element of the buffer.
+  inline T& peekFrontMut() noexcept {
+    return buffer_[headIndex_];
+  }
+
+  /// @brief Peek at the back of the buffer.
+  /// @return A mutable reference to the back element of the buffer.
+  inline T& peekBackMut() noexcept {
+    return buffer_[prevIndex(tailIndex_)];
+  }
+
+  /// @brief Check if the buffer is empty.
+  /// @return True if the buffer is empty, false otherwise.
+  const inline bool isEmpty() const noexcept {
+    return headIndex_ == tailIndex_;
+  }
+
+  /// @brief Check if the buffer is full.
+  /// @return True if the buffer is full, false otherwise.
+  const inline bool isFull() const noexcept {
+    return nextIndex(tailIndex_) == headIndex_;
+  }
+
+  /// @brief Get the capacity of the buffer.
+  /// @return The capacity of the buffer.
+  const inline size_t getCapacity() const noexcept {
+    return capacity_;
+  }
+
+  /// @brief Get the real capacity of the buffer (excluding one slot for the empty state).
+  /// @return The real capacity of the buffer.
+  const inline size_t getRealCapacity() const noexcept {
+    return capacity_ - 1;
+  }
+
+  /// @brief Get the number of elements in the buffer.
+  /// @return The number of elements in the buffer.
+  const inline size_t size() const noexcept {
+    return (capacity_ + tailIndex_ - headIndex_) & (capacity_ - 1);
+  }
```

Comment:
Top level const qualifiers are completely ignored here, remove them ;)
```suggestion
  inline bool isEmpty() const noexcept {
    return headIndex_ == tailIndex_;
  }

  /// @brief Check if the buffer is full.
  /// @return True if the buffer is full, false otherwise.
  inline bool isFull() const noexcept {
    return nextIndex(tailIndex_) == headIndex_;
  }

  /// @brief Get the capacity of the buffer.
  /// @return The capacity of the buffer.
  inline size_t getCapacity() const noexcept {
    return capacity_;
  }

  /// @brief Get the real capacity of the buffer (excluding one slot for the empty state).
  /// @return The real capacity of the buffer.
  inline size_t getRealCapacity() const noexcept {
    return capacity_ - 1;
  }

  /// @brief Get the number of elements in the buffer.
  /// @return The number of elements in the buffer.
  inline size_t size() const noexcept {
    return (capacity_ + tailIndex_ - headIndex_) & (capacity_ - 1);
  }
```